File: strings/src/test/com/whiteboard/FindSubstringTest.java
package java.com.whiteboard;

import org.junit.Test;

public class FindSubstringTest {
    @Test
    public void findSubstringTest() {
        System.out.println();
        System.out.println("Test FindSubstring:");
        System.out.println("===================");

        String testString = "alfalfa romeo";
        String[] testSubstrings = {
                "alfa",
                "rome",
                "alfa romeo",
                "alfa romeos",
                "alfalfa romeo",
                "alfalfa romeos",
                "x"
        };

        System.out.println("Test string = " + testString);
        for (String s : testSubstrings) {
            int index = FindSubstring.findSubstring(testString, s);
            assert(index == testString.indexOf(s));
            System.out.println("findSubstring(" + s + ") = " + index);
        }
    }
}

File: strings/src/test/com/whiteboard/kotlin/FindSubstringTest.kt
package com.whiteboard.kotlin

import com.whiteboard.java.FindSubstring
import org.junit.Test

class FindSubstringTest {
    @Test
    fun findSubstringTest() {
        println()
        println("Test FindSubstring:")
        println("===================")
        val testString = "alfalfa romeo"
        val testSubstrings = arrayOf(
            "alfa",
            "rome",
            "alfa romeo",
            "alfa romeos",
            "alfalfa romeo",
            "alfalfa romeos",
            "x"
        )
        println("Test string = $testString")
        for (s in testSubstrings) {
            val index = FindSubstring.findSubstring(testString, s)
            assert(index == testString.indexOf(s))
            println("findSubstring($s) = $index")
        }
    }
}